akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/HeaderDirectivesSpec.scala

```diff
@@ -202,4 +206,30 @@ class HeaderDirectivesSpec extends RoutingSpec with Inside {
       }
     }
   }
+
+  "The HeaderMagnet" should {
```

Comment:
HeaderMagnet is only an implementation detail, so not sure if we need exactly those tests. Could you additionally (or instead) create one or more tests, e.g. above for `headerValueForType` that asserts whether the rejection is returned similar to the test you changed in the ModeledCustomHeaderSpec?

Reply:
Yeah. That should be easy enough. I'll do that when I get some free time, maybe later this week.